internal/api/dto/invoice.go

```diff
@@ -17,8 +19,8 @@ type CreateInvoiceRequest struct {
 	CustomerID     string                         `json:"customer_id" validate:"required"`
 	SubscriptionID *string                        `json:"subscription_id,omitempty"`
 	IdempotencyKey *string                        `json:"idempotency_key"`
-	InvoiceType    types.InvoiceType              `json:"invoice_type" validate:"required"`
-	Currency       string                         `json:"currency" validate:"required"`
+	InvoiceType    types.InvoiceType              `json:"invoice_type" validate:"oneof=SUBSCRIPTION ONE_OFF CREDIT"`
+	Currency       string                         `json:"currency" validate:"required,iso4217"`
```

Comment:
these validations were missing

instead of this we can also add [this](https://github.com/go-playground/validator/blob/master/_examples/struct-level/main.go#L81) - registering struct validations